code/modules/reagents/chemistry/holder.dm

```diff
@@ -408,9 +410,11 @@
 	if(can_overdose)
 		if(addiction_tick == 6)
 			addiction_tick = 1
-			for(var/addiction in cached_addictions)
-				var/datum/reagent/R = addiction
-				if(C && R)
+			if(cached_addictions)
```

Comment:
```suggestion
```
Any reason for this `if` check? `for(var/i in null)` will properly exit correctly without runtiming, no need to check if the list exists.